eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java

```diff
@@ -149,104 +178,158 @@ public EclipseCommitValidationListener(PluginConfigFactory cfgFactory) {
 			if (hasCurrentAgreement(authorIdent, author)) {
-				messages.add(new CommitValidationMessage("The author has a current Eclipse Contributor Agreement (ECA) on file.", false));	
+				messages.add(new CommitValidationMessage(
+						"The author has a current Eclipse Contributor Agreement (ECA) on file.", false));
 			} else {
-				messages.add(new CommitValidationMessage("The author does not have a current Eclipse Contributor Agreement (ECA) on file.\n" + 
-				"If there are multiple commits, please ensure that each author has a ECA.", true));	
+				messages.add(
+						new CommitValidationMessage(
+								"The author does not have a current Eclipse Contributor Agreement (ECA) on file.\n"
+										+ "If there are multiple commits, please ensure that each author has a ECA.",
+								true));
 				addEmptyLine(messages);
 				errors.add("An Eclipse Contributor Agreement is required.");
 			}
-			
+
 			if (hasSignedOff(authorIdent, author, commit)) {
 				messages.add(new CommitValidationMessage("The author has \"signed-off\" on the contribution.", false));
 			} else {
-				messages.add(new CommitValidationMessage("The author has not \"signed-off\" on the contribution.\n" +
-					"If there are multiple commits, please ensure that each commit is signed-off.", true));
+				messages.add(new CommitValidationMessage(
+						"The author has not \"signed-off\" on the contribution.\n"
+								+ "If there are multiple commits, please ensure that each commit is signed-off.",
+						true));
 				errors.add("The contributor must \"sign-off\" on the contribution.");
 			}
-			
+
 			// TODO Extend exception-throwing delegation to include all possible messages.
 			if (!errors.isEmpty()) {
 				addDocumentationPointerMessage(messages);
 				throw new CommitValidationException(errors.get(0), messages);
 			}
 		}
-		
+
 		addEmptyLine(messages);
-		
+
 		/*
-		 * Only committers can push on behalf of other users. Note that, I am asking
-		 * if the user (i.e. the person who is doing the actual push) is a committer.
+		 * Only committers can push on behalf of other users. Note that, I am asking if
+		 * the user (i.e. the person who is doing the actual push) is a committer.
 		 */
-		if (!author.isPresent() || !author.get().getAccount().equals(user.getAccount())) {
-			if (!isCommitter(user, project)) {
-				messages.add(new CommitValidationMessage("You are not a project committer.", true));
-				messages.add(new CommitValidationMessage("Only project committers can push on behalf of others.", true));
-				addDocumentationPointerMessage(messages);
-				addEmptyLine(messages);
-				throw new CommitValidationException("You must be a committer to push on behalf of others.", messages);	
-			}
-		} 
+		if ((!author.isPresent() || !author.get().getAccount().equals(user.getAccount()))
+				&& (!isCommitter(user, project) && (!author.isPresent()
+						|| !getPreviousPatchsetAuthors(project, commit).contains(author.get())))) {
```

Comment:
This is unreadable. Mathias had 3 nested `if` that were much more understandable. Please revert.